### PR TITLE
Add support of snap command

### PR DESCRIPTION
**Pull request by vadmeste**
Add snapshot command implement: mc snap -h for full command list.

### CODE REVIEW

File: cmd/snap-serialize.go
Comment (klauspost) [inline cmd/snap-serialize.go]: @vadmeste The context provides that functionality.

Comment (vadmeste) [reply cmd/snap-serialize.go]: > @vadmeste The context provides that functionality.

The problem with context is that it always provides an error, but I want from BucketEntries to return nil in that case.

**Comment by vadmeste** [comment]
Not WIP anymore.

**Comment by klauspost** [comment]
@vadmeste Could you merge master? I will add my review.

File: cmd/client-snapshot.go
Comment (klauspost) [inline cmd/client-snapshot.go:545]:
```suggestion
		API:     "GetVersioning",
```
Couldn't this be supported?

File: cmd/client-snapshot.go
Comment (klauspost) [inline cmd/client-snapshot.go:553]:
```suggestion
		API:     "SetVersioning",
```

File: cmd/snap-main.go
Comment (klauspost) [inline cmd/snap-main.go:31]:
```suggestion
	Usage:           "generate snapshots of S3 object metadata",
```

File: cmd/snap-remove.go
Comment (klauspost) [inline cmd/snap-remove.go:38]: Do we really need to ask? It is 'only' a snapshot.

Just drop the confirmation IMO.

File: docs/minio-client-complete-guide.md
Comment (klauspost) [inline docs/minio-client-complete-guide.md:869]:
```suggestion
`snap` command generates snapshots of S3 metadata at a given date/time. It relies on S3 bucket versioning feature to pick objects versions.
```

We do not in any way want to indicate that it stores actual data (for now at least).

Comment (klauspost) [reply docs/minio-client-complete-guide.md:869]: Could you add a sentence on what it can be used for?

**Review by klauspost — COMMENTED** [review]
Nothing big.

**Comment by harshavardhana** [comment]
Are we ready for this now? @vadmeste

File: cmd/snap-remove.go
Comment (vadmeste) [reply cmd/snap-remove.go:38]: > Do we really need to ask? It is 'only' a snapshot.
> 
> Just drop the confirmation IMO.

Let's wait until this is approved/disapproved by AB.

I always want confirmation for delete.

**Comment by vadmeste** [comment]
I rebased this PR and added some fixes. It is ready for review now.

File: cmd/snap-remove.go
Comment (klauspost) [reply cmd/snap-remove.go:38]: `mc rm file` doesn't require confirmation. 

We are not deleting actual data, so I don't really see why it is so important.

**Comment by klauspost** [comment]
@vadmeste Needs rebase

Are we only waiting for review of this one, or has it deliberately been put on hold?

**Comment by klauspost** [comment]
I think we should offer to snapshot file contents as well, but this could rather easily be extended to include that at some point.

File: cmd/client-s3.go
Comment (kannappanr) [inline cmd/client-s3.go:2]:
```suggestion
 * MinIO Client (C) 2020 MinIO, Inc.
```
```suggestion
 * MinIO Client (C) 2015-2020 MinIO, Inc.
```

File: cmd/config.go
Comment (kannappanr) [inline cmd/config.go:311]: Please add comments if relevant

**Review by klauspost — APPROVED** [review]

**Comment by harshavardhana** [comment]
Let's revisit this later when we have time - closing for now.

**Pull request closed without merging**
